### PR TITLE
fix(TDI-40936):Kinesis OOM

### PR DESCRIPTION
**Pull request by pyzhou-talend**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-40936

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/72/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/72/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/77/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/77/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/79/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/79/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Apyzhou%2FTDI-40936-kinesis-OOM)

File: components/components-kinesis/kinesis-runtime/src/main/java/org/talend/components/kinesis/runtime/KinesisDatasetRuntime.java [hunk 1]
```diff
@@ -88,7 +88,6 @@ public void getSample(int limit, Consumer<IndexedRecord> consumer) {
             // Collect a sample of the input records.
             p
                     .apply(inputRuntime) //
-                    .apply(Sample.<IndexedRecord> any(limit))
```
Comment (wwang-talend) [inline components/components-kinesis/kinesis-runtime/src/main/java/org/talend/components/kinesis/runtime/KinesisDatasetRuntime.java]: if we remove the code here, no place to limit the returned row number.

Comment (pyzhou-talend) [reply components/components-kinesis/kinesis-runtime/src/main/java/org/talend/components/kinesis/runtime/KinesisDatasetRuntime.java]: if the code here does not limit the amount, I don't know where to do that.
https://github.com/Talend/components/blob/96d78392b15521591f96e101b88fe0e777344a31/components/components-kinesis/kinesis-runtime/src/main/java/org/talend/components/kinesis/runtime/KinesisDatasetRuntime.java#L77

**Pull request closed without merging**